parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -1017,6 +987,52 @@ pub mod pallet {
 				},
 			}
 		}
+
+		pub fn participant_pubkeys(
+			sync_committee_bits: &[u8],
+			sync_committee_pubkeys: &Vec<PublicKeyPrepared>,
+		) -> Result<Vec<PublicKeyPrepared>, Error<T>> {
+			let mut pubkeys: Vec<PublicKeyPrepared> = Vec::new();
+			for (bit, pubkey) in sync_committee_bits.iter().zip(sync_committee_pubkeys.iter()) {
+				if *bit == 1 as u8 {
+					let pubk = pubkey.clone();
+					pubkeys.push(pubk);
+				}
+			}
+			Ok(pubkeys)
+		}
+
+		pub fn absent_pubkeys(
+			sync_committee_bits: &[u8],
+			sync_committee_pubkeys: &Vec<PublicKeyPrepared>,
+		) -> Result<Vec<PublicKeyPrepared>, Error<T>> {
```

Comment:
ditto:

```suggestion
		) -> Vec<PublicKeyPrepared> {
```

Reply:
Also both `participant_pubkeys` and `absent_pubkeys` are almost exact duplicates. Lets try to merge them into a single function that takes a `fn(u8) -> bool` function argument to decide which keys to filter.